astropy/coordinates/builtin_frames/galactocentric.py

```diff
@@ -82,100 +102,231 @@ class galactocentric_frame_defaults(ScienceState):
 
-        # Resolve the meaning of 'latest': The latest parameter set is from v4.0
+        """
+        # Resolve the meaning of 'latest': latest parameter set is from v4.0
         # - update this as newer parameter choices are added
-        if arg == 'latest':
-            arg = cls._latest_value
-
-        params = dict()
-        references = dict()
-
-        # Currently, all versions use the same sky position for Sgr A*:
-        params['galcen_coord'] = ICRS(ra=266.4051*u.degree,
-                                      dec=-28.936175*u.degree)
-        references['galcen_coord'] = \
-            'http://adsabs.harvard.edu/abs/2004ApJ...616..872R'
-
-        # The roll angle is the same for both frames:
-        params['roll'] = 0 * u.deg
-
-        if arg == 'pre-v4.0':
-            params['galcen_distance'] = 8.3 * u.kpc
-            references['galcen_distance'] = \
-                'https://ui.adsabs.harvard.edu/#abs/2009ApJ...692.1075G'
-
-            params['galcen_v_sun'] = r.CartesianDifferential([11.1,
-                                                              220+12.24,
-                                                              7.25]*u.km/u.s)
-            references['galcen_v_sun'] = \
-                ['https://ui.adsabs.harvard.edu/#abs/2010MNRAS.403.1829S',
-                 'https://ui.adsabs.harvard.edu/#abs/2015ApJS..216...29B']
-
-            params['z_sun'] = 27.0 * u.pc
-            references['z_sun'] = \
-                'https://ui.adsabs.harvard.edu/#abs/2001ApJ...553..184C'
-
-        elif arg == 'v4.0':
-            params['galcen_distance'] = 8.122 * u.kpc
-            references['galcen_distance'] = \
-                'https://ui.adsabs.harvard.edu/abs/2018A%26A...615L..15G'
-
-            params['galcen_v_sun'] = r.CartesianDifferential([12.9,
-                                                              245.6,
-                                                              7.78]*u.km/u.s)
-            references['galcen_v_sun'] = \
-                ['https://ui.adsabs.harvard.edu/abs/2018RNAAS...2..210D',
-                 'https://ui.adsabs.harvard.edu/abs/2018A%26A...615L..15G',
-                 'https://ui.adsabs.harvard.edu/abs/2004ApJ...616..872R']
-
-            params['z_sun'] = 20.8 * u.pc
-            references['z_sun'] = \
-                'https://ui.adsabs.harvard.edu/abs/2019MNRAS.482.1417B'
+        if name == 'latest':
+            name = cls._latest_value
 
-        else:
-            raise ValueError(f'Invalid string input to retrieve solar '
-                             f'parameters for Galactocentric frame: "{arg}"')
+        # Get the state from the registry.
+        # Copy to ensure registry is immutable to modifications of "_value".
+        # Raises KeyError if `name` is invalid string input to registry
+        # to retrieve solar parameters for Galactocentric frame.
+        state = copy.deepcopy(cls._registry[name])  # ensure mutable
```

Comment:
I'm not quite sure why we do not just return `state` here. Up to the user to deal with the absence of references, I'd think. And for `parameters`, it is unclear to me that setting it to `None` if they do not exist is helpful...

Reply:
✔︎
It was, admittedly, overly cautious. I'll let the user live or die by their own registration 🙃.